### DISCUSSION TITLE
How to handle Google calendar in Chrome for iOS?

### DISCUSSION THREAD

**Discussion by georgiarnaudov** [add2cal/add-to-calendar-button#418] [Q&A]
Hello, I may be repeating the question, but couldn't find how to make a better experience for users on iOS which are using Chrome. 

The main issue is that they need to make an additional step after the instructions pop.

Is there any solution to this or that is a limitation of webviews at the moment?

**Comment by add-to-calendar** [comment]
It is a current limitation of Chrome on iOS when delivering the ics file dynamically.
You can circumvent this by providing a static ics file via the "icsFile" option (providing a url to the ics file on your server). In doubt, click the button yourself to get the ics file, upload the ics file to your server, get its URL, set this URL with the "icsFile" option.

We are currently working on a managed solution, which does this for you. For now, the only way would be the "icsFile" option.

**Reply by georgiarnaudov** [reply]
I see, thanks a lot for the clarification! Nice to hear you're doing a managed solution. Will keep an eye.

**Reply by georgiarnaudov** [reply]
@add-to-calendar Implemented the suggested solution, but whenever you click add to calendar, Chrome asks you to save the file. Then when you save it, there's no suggestion to open the file with the Calendar app. If I try opening it while in the Files app, nothing happens. Really odd, is that whats expected? :)

**Reply by jekuer** [reply]
Unfortunately, yes.
This is rather another ugly shortcoming of iOS. They do not properly support ics files on a file level. Technically, everything is how it should be. It is simply Apple in the end, not being able to open a calendar file with the calendar app.

We are currently discussing whether we disable ics files on iOS generally. Would mean that the "Apple" and "iCal" options would only work in Safari. However, also somehow fuzzy to cover up Apple's fuc*-up this way.
We will optimize a little bit in terms of user guidance (on iOS, Apple option will be forced to iCal), but not yet sure about more.


See e.g. https://discussions.apple.com/thread/254619998 or many many other Apple support rants.

**Reply by georgiarnaudov** [reply]
Doh, Apple being Apple 😀🙉. Still we're 2023 and these kind of things block us doing our job. 

In my case is an easy switch because I will do the project with ReactNative and use native packages for saving events into the calendar.

Thank you for taking the time to answer!

**Reply by jekuer** [reply]
we might also simply keep showing the note for Chrome on iOS and extend it to all cases. Might be the best UX. Besides, only around 5% of iPhone users are using Chrome. Therefore, still a rather minor problem.

**Reply by add-to-calendar** [reply]
Behavior changed with latest release. Static ics now also shows the note. Hopefully we can update this for the better in the future.

**Comment by jekuer** [comment]
The latest version comes with an additional update. For Chrome on iOS, if you have a static ics file given, we now force the subscribe mdoe via the webcal protocol. Not ideal, but it works for the user and is more straightforward (also the way AddEvent does it, for example).
Best approach for now. Hopefully, there will be better support by Apple and/or Google in the future.
